stdlib/public/runtime/ImageInspectionCommon.cpp

```diff
@@ -138,6 +137,52 @@ void *swift::lookupSection(const char *segment, const char *section, size_t *out
   return nullptr;
 }
 
+#ifndef NDEBUG
+SWIFT_RUNTIME_EXPORT
+const swift::MetadataSections *swift_getMetadataSection(size_t index) {
+  if (swift::registered == nullptr) {
+    return nullptr;
+  }
+
+  auto selected = swift::registered;
+  while (index > 0) {
+    selected = selected->next;
+    if (selected == swift::registered) {
+      return nullptr;
+    }
+    --index;
+  }
+  return selected;
+}
+
+SWIFT_RUNTIME_EXPORT
+const char *swift_getMetadataSectionName(void *metadata_section) {
+  swift::SymbolInfo info;
+  if (lookupSymbol(metadata_section, &info)) {
+    if (info.fileName) {
+      return info.fileName;
+    }
+  }
+  return "Anonymous";
```

Comment:
For ELF at least, the empty string is traditionally used to indicate the section name cannot be identified.